libc/unistd_fsdir.c

```diff
@@ -75,41 +82,58 @@ TEST(unistd_fsdir, getcwd)
 }
 
 
-TEST(unistd_fsdir, chdir_abs)
+TEST(unistd_fsdir, chdir_absroot)
 {
-	/* test chdir to some directory */
-	TEST_ASSERT_EQUAL_INT(0, chdir("/dev"));
+	/* test chdir to root */
+	TEST_ASSERT_EQUAL_INT(0, chdir("/"));
 	TEST_ASSERT_NOT_NULL(getcwd(buf, sizeof(buf)));
-	TEST_ASSERT_EQUAL_STRING("/dev", buf);
+	TEST_ASSERT_EQUAL_STRING("/", buf);
+}
 
-	/* test chdir to cwd */
+
+TEST(unistd_fsdir, chdir_absdev)
+{
+	/* test chdir to some directory */
 	TEST_ASSERT_EQUAL_INT(0, chdir("/dev"));
 	TEST_ASSERT_NOT_NULL(getcwd(buf, sizeof(buf)));
 	TEST_ASSERT_EQUAL_STRING("/dev", buf);
-
-	/* test chdir back to root */
-	TEST_ASSERT_EQUAL_INT(0, chdir("/"));
-	TEST_ASSERT_NOT_NULL(getcwd(buf, sizeof(buf)));
-	TEST_ASSERT_EQUAL_STRING("/", buf);
 }
 
 
 TEST(unistd_fsdir, chdir_rel)
 {
+	char absPath[PATH_MAX];
+	size_t slen, free;
+
+	strncpy(absPath, testWorkDir, sizeof(absPath));
+
+	slen = strlen(absPath);
+	if (absPath[slen - 1] != '/') {
+		free = (size_t)sizeof(absPath) - slen;
+		strncat(absPath, "/", free);
+	}
+	slen = strlen(absPath);
+	free = (size_t)sizeof(absPath) - slen;
```

Comment:
if `free` is too smallYou will get strange DIRNAME and the whole test would fail. Also, `strcat` is definitely not optimal.

maybe:
```c
size_t slen = strlen(absPath);
TEST_ASSERT_GREATER_EQUAL(sizeof(absPath), slen + sizeof(DIRNAME) + 2);  // or sth like that 
if (absPath[slen - 1] != '/') {
  absPath[slen++] = '/';
  absPath[slen++] = '\0';
}
strcpy(absPath + slen, DIRNAME);